src/services/tokenService.js

```diff
@@ -0,0 +1,28 @@
+const Token = require('../models/token.js');
+
+const save = async (userId, newToken) => {
+  const token = await Token.findOne({ where: { userId } });
+
+  if (!token) {
+    await Token.create({ userId, refreshToken: newToken });
+
+    return;
+  }
+
+  token.refreshToken = newToken;
+  await token.save();
```

Comment:
Ensure that `await token.save()` is wrapped in a try-catch block to handle any potential errors during the save operation. This will prevent unhandled promise rejections.